core/config/toml/types.go

```diff
@@ -1693,11 +1693,8 @@ func (b *Telemetry) ValidateConfig() (err error) {
 	if b.Endpoint == nil || *b.Endpoint == "" {
 		err = multierr.Append(err, configutils.ErrMissing{Name: "Endpoint", Msg: "must be set when Telemetry is enabled"})
 	}
-	if b.InsecureConnection != nil && *b.InsecureConnection {
-		if build.IsProd() {
-			err = multierr.Append(err, configutils.ErrInvalid{Name: "InsecureConnection", Value: true, Msg: "cannot be used in production builds"})
-		}
-	} else {
+	if b.InsecureConnection != nil && !*b.InsecureConnection {
```

Comment:
Shouldn't this still apply for nill too?
```suggestion
	if b.InsecureConnection == nil || !*b.InsecureConnection {
```

Reply:
I thought the nil checks were just for type safety? At this point aren't all the values initialized to their defaults (i.e. `false` )

Reply:
I don't think this code should be making any assumptions about the broader system. Before this change, we had an `if !nil && true` making the else case `nil || false`. The minimal impact change would be to continue to match `nil || false`. Dropping the nil case introduces risk without reward.